app/test/features/chat_ng/widgets/chat_messages_list_test.dart

```diff
@@ -0,0 +1,406 @@
+        overrides: [
+          chatMessagesStateProvider(
+            testRoomId,
+          ).overrideWith((ref) => MockChatRoomMessagesNotifier(messagesState)),
+          chat
+              .timelineStreamProvider(testRoomId)
+              .overrideWith((ref) => Future.value(MockTimelineStream())),
+          animatedListChatMessagesProvider(
+            testRoomId,
+          ).overrideWith((ref) => GlobalKey<AnimatedListState>()),
+        ],
+        child: ChatMessages(roomId: testRoomId),
+      );
+
+      final scrollController =
+          tester.widget<AnimatedList>(find.byType(AnimatedList)).controller;
+      final chatMessagesWidget = find.byType(ChatMessages);
+
+      // scroll up to make the button visible
+      scrollController?.position.jumpTo(100.0);
+
+      await tester.pump();
+
+      final initialState =
+          tester.state(chatMessagesWidget) as ConsumerState<ChatMessages>;
+      expect((initialState as dynamic).showScrollToBottom, isTrue);
+
+      final finalState =
+          tester.state(chatMessagesWidget) as ConsumerState<ChatMessages>;
+
+      await (finalState as dynamic).scrollToEnd();
+
+      await tester.pump();
+      //wait for the animation to complete
+      await tester.pumpAndSettle();
+
+      expect(scrollController?.position.pixels, equals(0.0));
+      // button should be hidden
+      expect((finalState as dynamic).showScrollToBottom, isFalse);
+    });
+
+    testWidgets('preserves correct message order when displayed in reverse', (
+      tester,
+    ) async {
+      final messagesState = ChatRoomState(
+        messageList: List.generate(5, (index) => 'msg$index'),
+        loading: const ChatRoomLoadingState.loaded(),
+      );
+
+      final mockNotifier = MockChatRoomMessagesNotifier(messagesState);
+
+      await tester.pumpProviderWidget(
+        overrides: [
+          chatMessagesStateProvider(
+            testRoomId,
+          ).overrideWith((ref) => mockNotifier),
+          chat
+              .timelineStreamProvider(testRoomId)
+              .overrideWith((ref) => Future.value(MockTimelineStream())),
+          animatedListChatMessagesProvider(
+            testRoomId,
+          ).overrideWith((ref) => GlobalKey<AnimatedListState>()),
+          isActiveProvider(LabsFeature.chatUnread).overrideWith((ref) => false),
+        ],
+        child: ChatMessages(roomId: testRoomId),
+      );
+
+      await tester.pump();
+
+      // Get the AnimatedList
+      final animatedList = tester.widget<AnimatedList>(
+        find.byType(AnimatedList),
+      );
+
+      // Verify the list is in reverse
+      expect(animatedList.reverse, isTrue);
+
+      // Verify the correct number of items
+      expect(animatedList.initialItemCount, equals(5));
+
+      // Check the order of messages in the state
+      expect(
+        mockNotifier.state.messageList,
+        equals(['msg0', 'msg1', 'msg2', 'msg3', 'msg4']),
+      );
+    });
+  });
+}
```

Comment:
so close! the chat room message notifier in codecov just complains about one line never being met, and that is removed items. Maybe we can add a tests for removing items as well?